src/main/java/de/tum/cit/aet/artemis/programming/service/localvc/LocalVCRepositoryUri.java

```diff
@@ -31,26 +33,41 @@ public class LocalVCRepositoryUri extends VcsRepositoryUri {
      *
+     * @param localVCBaseUri The base URI of the local version control server, provided as a {@link URI}, which forms the prefix of the full repository URI.
      * @param projectKey     The project key that identifies the group or project under which the repository is categorized in the local VC system.
      * @param repositorySlug The slug or name of the repository, typically used as a unique identifier for the repository within the project.
-     * @param localVCBaseUri The base URI of the local version control server, provided as a {@link URI}, which forms the prefix of the full repository URI.
      * @throws LocalVCInternalException If the project key or repository slug results in an invalid URI, encapsulating the {@link URISyntaxException}.
      */
-    public LocalVCRepositoryUri(String projectKey, String repositorySlug, URI localVCBaseUri) {
-        final String uriString = localVCBaseUri + buildRepositoryPath(projectKey, repositorySlug);
-        try {
-            this.uri = new URI(uriString);
-        }
-        catch (URISyntaxException e) {
-            throw new LocalVCInternalException("Could not create local VC Repository URI", e);
-        }
-
+    public LocalVCRepositoryUri(URI localVCBaseUri, String projectKey, String repositorySlug) {
+        this.uri = buildUri(localVCBaseUri, projectKey, repositorySlug);
         this.projectKey = projectKey;
         this.repositorySlug = repositorySlug;
         this.repositoryTypeOrUserName = getRepositoryTypeOrUserName(repositorySlug, projectKey);
         this.isPracticeRepository = isPracticeRepository(repositorySlug, projectKey);
     }
 
+    private static URI buildUri(URI localVCBaseUri, String projectKey, String repositorySlug) {
+        return UriComponentsBuilder.fromUri(localVCBaseUri).pathSegment("git", projectKey, repositorySlug + ".git").build().toUri();
+    }
+
+    /**
+     * Initializes a new instance of the {@link LocalVCRepositoryUri} class from a repository name
+     * and builds an url to format <code>{server.url}/git/{project-key}/{repo-name}.git</code> with <code>repo-name</code> consisting of <code>{project-key}-{repo-type}</code>
+     *
+     * @param localVcBaseUri The base URI of the local version control system
+     * @param repositoryName containing the project key at the beginning, e.g. {project-key}-{repo-name}
+     */
+    public LocalVCRepositoryUri(URI localVcBaseUri, String repositoryName) throws URISyntaxException {
+        this(localVcBaseUri, extractProjectKey(repositoryName), repositoryName);
+    }
+
+    private static String extractProjectKey(String repositoryName) {
+        if (!repositoryName.matches("[a-zA-Z0-9]+-[a-zA-Z0-9-]+")) {
+            throw new IllegalArgumentException("Repository name must be in the format <project-key>-<repo-type>");
+        }
+        return repositoryName.split("-")[0];
+    }
```

Comment:
I think it the error message would be even easier to read like that
```suggestion
    private static String extractProjectKey(String repositoryName) {
        if (!repositoryName.matches("[a-zA-Z0-9]+-[a-zA-Z0-9-]+")) {
            throw new IllegalArgumentException("Repository name must be in the format <projectKey>-<repoType>");
        }
        return repositoryName.split("-")[0];
    }
```